pkg/storage/storage.go

```diff
@@ -40,6 +40,10 @@ func CheckConfigMapEntry(key string, ins types.NamespacedName) (string, error) {
 		return "", err
 	}
 
+	if !found {
```

Comment:
don't we want to return an error in case ConfigMap does not have a "data" field? It means that it is invalid ConfigMap, no?